alma/tests/Unit/Helper/InsuranceProductHelperTest.php

```diff
@@ -0,0 +1,163 @@
+use PHPUnit\Framework\TestCase;
+
+class InsuranceProductHelperTest extends TestCase
+{
+    /**
+     * @var InsuranceProductHelper
+     */
+    protected $insuranceProductHelper;
+    /**
+     * @var AlmaInsuranceProductRepository|(AlmaInsuranceProductRepository&\PHPUnit_Framework_MockObject_MockObject)|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $almaInsuranceProductRepository;
+    /**
+     * @var \Cart|(\Cart&\PHPUnit_Framework_MockObject_MockObject)|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $cart;
+    /**
+     * @var \Cart|(\Cart&\PHPUnit_Framework_MockObject_MockObject)|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $newCart;
+    /**
+     * @var \Mockery\Mock|(\Mockery\MockInterface&InsuranceProductHelperBuilder)
+     */
+    private $insuranceProductHelperBuilderMock;
+
+    protected function setUp()
+    {
+        $this->insuranceProductHelperBuilderMock = \Mockery::mock(InsuranceProductHelperBuilder::class)->makePartial();
+        $this->insuranceProductHelper = $this->insuranceProductHelperBuilderMock->getInstance();
+        $this->almaInsuranceProductRepository = \Mockery::mock(AlmaInsuranceProductRepository::class)->makePartial();
+
+        $this->context = \Mockery::mock(\Context::class);
+        $this->contextFactory = \Mockery::mock(ContextFactory::class)->makePartial();
+        $this->contextFactory->shouldReceive('getContext')->andReturn($this->context);
+
+        $this->cart = $this->createMock(\Cart::class);
+        $this->newCart = $this->createMock(\Cart::class);
+        $this->shop = $this->createMock(\Shop::class);
+    }
+
+    /**
+     * @throws \PrestaShopDatabaseException
+     */
+    public function testDuplicateInsuranceProductsWithOneInsuranceProduct()
+    {
+        $almaInsuranceProduct = [[
+            'id_product' => '1',
+            'id_product_attribute' => '1',
+            'id_customization' => '0',
+            'id_product_insurance' => '20',
+            'id_product_attribute_insurance' => '42',
+            'id_address_delivery' => '7',
+            'price' => '2150',
+            'insurance_contract_id' => 'insurance_contract_3CR76PNE04X7hHEY7V65rZ',
+            'cms_reference' => '1-1',
+            'product_price' => '35000',
+        ]];
+        $this->cart->id = 1;
+        $this->newCart->id = 2;
+        $this->shop->id = 1;
+        $this->context->shop = $this->shop;
+
+        $this->almaInsuranceProductRepositorySpy = \Mockery::spy(AlmaInsuranceProductRepository::class)->makePartial();
+
+        $this->almaInsuranceProductRepositorySpy->shouldReceive('getByCartIdAndShop')
+            ->with($this->cart->id, $this->context->shop->id)
+            ->andReturn($almaInsuranceProduct);
+
+        $this->almaInsuranceProductRepositorySpy->shouldReceive('add')->andReturn(null);
+
+        $this->insuranceProductHelper = \Mockery::mock(InsuranceProductHelper::class, [
```

Comment:
you should not make a mock if it is not necessary for the class on which you are testing